dhall/src/Dhall/Import/Types.hs

```diff
@@ -39,9 +41,12 @@ data Status m = Status
     -- ^ Stack of `Import`s that we've imported along the way to get to the
     -- current point
 
-    , _cache :: Map Import (Expr Src X)
-    -- ^ Cache of imported expressions in order to avoid importing the same
-    --   expression twice with different values
+    , _dot :: (Int, Dot NodeId)
```

Comment:
Rather than use `_dot :: (Int, Dot NodeId)` I would suggest two separate fields, such as `_dot :: Dot NodeId` and `_nodeId :: Int`